Sources/MongoSwift/Operations/WatchOperation.swift

```diff
@@ -31,32 +33,34 @@ internal struct WatchOperation<CollectionType: Codable, ChangeStreamType: Codabl
         let pipeline: Document = ["pipeline": self.pipeline]
         let opts = try encodeOptions(options: self.options, session: session)
 
+        let changeStream: OpaquePointer
+        let client: MongoClient
+        let decoder: BSONDecoder
+
         switch self.target {
-        case let .client(client):
-            return try ChangeStream<ChangeStreamType>(options: self.options,
-                                                      client: client,
-                                                      decoder: client.decoder,
-                                                      session: session) { conn in
-                mongoc_client_watch(conn.clientHandle, pipeline._bson, opts?._bson)
-            }
+        case let .client(c):
+            client = c
+            decoder = c.decoder
+            changeStream = mongoc_client_watch(connection.clientHandle, pipeline._bson, opts?._bson)
```

Comment:
why is this one different than collection and database?

Reply:
sorry, I misread this to be assigning a changestream type rather than a pointer!

Reply:
sorry, I misread this to be assigning a changestream type rather than a pointer!